src/FormatNumber.elm

```diff
@@ -37,115 +95,61 @@ type alias Locale =
     >>> formatFloat (Locale 1 "," ".") -0.01
     "0.0"
 
+    >>> formatFloat (Locale 2 "," ".") 0.01
+    "0.01"
+
+    >>> formatFloat (Locale 0 "," ".") 123.456
+    "123"
+
+    >>> formatFloat (Locale 0 "," ".") 1e9
+    "1,000,000,000"
+
+    >>> formatFloat (Locale 5 "," ".") 1.0
+    "1.00000"
 -}
 formatFloat : Locale -> Float -> String
 formatFloat locale num =
-    let
-        multiplier : Int
-        multiplier =
-            10 ^ locale.decimals
-
-        digits : Int
-        digits =
-            num
-                |> (*) (toFloat multiplier)
-                |> round
-    in
-        formattedNumber locale digits
+    (formatInt locale (truncate num))
+        ++ (separator locale)
+        ++ (nDigits locale.decimals num)
 
 
 {-| Format a integer number as a pretty string:
 
     >>> formatInt { decimals = 1, thousandSeparator = ",", decimalSeparator = "." } 0
-    "0.0"
+    "0"
```

Comment:
If your program  manipulates amounts of money, it has to be done with either a float or an int representing **the cents**. I can't think of any good reason one would store an amount of money in dollars as an `Int`.

So an user would use the API this way:

```elm
type alias Cents = Int

displayDollars : Cents -> String
displayDollars cents = "$ " ++ (formatFloat {usLocale | decimals = 2} (toFloat(cents)/100))
```